components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationPaginationAndSearching.java

```diff
@@ -94,10 +94,28 @@ ApplicationBasicInfo[] getPaginatedApplicationBasicInfo(String tenantDomain, Str
      * @param limit        Counting value.
      * @return An array of {@link ApplicationBasicInfo} instances matching the given filter within the given limit.
      * @throws IdentityApplicationManagementException Error in retrieving basic application information.
+     * @Deprecated This is being deprecated as introducing excludeSystemPortals to exclude system portals in response
+     * use {@link ApplicationPaginationAndSearching#getApplicationBasicInfo(String, String, String, int, int, Boolean)}.
      */
     ApplicationBasicInfo[] getApplicationBasicInfo(String tenantDomain, String username, String filter, int offset,
                                                    int limit) throws IdentityApplicationManagementException;
 
+    /**
+     * Get all basic application information for a matching filter with pagination based on the offset and limit.
+     *
+     * @param tenantDomain          Tenant Domain.
+     * @param username              User name.
+     * @param filter                Application name filter.
+     * @param offset                Starting index of the count.
+     * @param limit                 Counting value.
+     * @param excludeSystemPortals  Exclude system portals.
+     * @return An array of {@link ApplicationBasicInfo} instances matching the given filter within the given limit.
+     * @throws IdentityApplicationManagementException Error in retrieving basic application information.
+     */
+    ApplicationBasicInfo[] getApplicationBasicInfo(String tenantDomain, String username, String filter, int offset,
```

Comment:
Should be a default method.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6026/commits/9ab5dadbb9c0b0f8ceffa6a997e49f282954572b